### ISSUE TITLE
To Future operator?

### ISSUE THREAD

**Issue by sdykae** [rxRust/rxRust#150]
When there is an observable sometimes we need to pass the subscribe next value to a return of a function.
```rust
let a = obserfable::of("".to_string);
```
from reading in https://docs.rs/futures/0.3.16/futures/
we need to implement our own future :( for this.

**Comment by iuslkae** [comment]
The way to unwrap the observable (implement a future) is quite unknown to me but from
https://users.rust-lang.org/t/can-you-turn-a-callback-into-a-future-into-async-await/49378/19

there is this slice of code that can help
https://github.com/arifd/miniquad_async/blob/master/src/main.rs

look at the implementation of FutureWithReturn  `just copy all`
then implement the way 

```rust
pub fn pollObservable<T>(o: ObservableBase<OfEmitter<T>>) -> FutureWithReturn<T> {
  let item = Rc::new(RefCell::new(None));
  {
    let item = item.clone();
    let b = o.subscribe(move |x| {
      *item.borrow_mut() = Some(x)
    });
  }
  FutureWithReturn { item }
}
```

then use like 
```rust
#[actix_web::main]
async fn main() {
  let a = observable::of(String::from("hello"));
  let fromA = pollObservable(a).await;
  println!("{}",fromA);
}
```
If someone find a better workaround please tag me :')

**Comment by M-Adoo** [comment]
There is not a to_future operator now, but it not hard to add it.

**Comment by M-Adoo** [comment]
closed by #217
